packages/auth/e2e/phone.e2e.js

```diff
@@ -39,4 +40,110 @@ describe('auth() => Phone', () => {
       // TODO test error code and message
     });
   });
+
+  describe('verifyPhoneNumber', async () => {
+    it('successfully verifies', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+      await firebase.auth().verifyPhoneNumber(TEST_PHONE_A, false, false);
+    });
+
+    it('uses the autoVerifyTimeout when a non boolean autoVerifyTimeoutOrForceResend is provided', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+      await firebase.auth().verifyPhoneNumber(TEST_PHONE_A, 0, false);
+    });
+
+    it('throws an error with an invalid on event', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+
+      try {
+        await firebase
+          .auth()
+          .verifyPhoneNumber(TEST_PHONE_A)
+          .on('example', () => {});
+
+        return Promise.reject(new Error('Did not throw Error.'));
+      } catch (e) {
+        e.message.should.containEql(
+          "firebase.auth.PhoneAuthListener.on(*, _, _, _) 'event' must equal 'state_changed'.",
+        );
+        return Promise.resolve();
+      }
+    });
+
+    it('throws an error with an invalid observer event', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+
+      try {
+        await firebase
+          .auth()
+          .verifyPhoneNumber(TEST_PHONE_A)
+          .on('state_changed', null, null, () => {});
+
+        return Promise.reject(new Error('Did not throw Error.'));
+      } catch (e) {
+        e.message.should.containEql(
+          "firebase.auth.PhoneAuthListener.on(_, *, _, _) 'observer' must be a function.",
+        );
+        return Promise.resolve();
+      }
+    });
+
+    it('successfully runs verification complete handler', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+
+      await firebase
+        .auth()
+        .verifyPhoneNumber(TEST_PHONE_A)
+        .then($ => $);
+
+      return Promise.resolve();
+    });
+
+    it('succesffuly runs and adds emiters', async () => {
+      const TEST_PHONE_A = '+447445255123';
+      const confirmResult = await firebase.auth().signInWithPhoneNumber(TEST_PHONE_A);
+
+      await confirmResult.confirm(TEST_CODE_A);
+
+      const obervserCb = () => {};
+
+      const errorCb = () => {};
+
+      const successCb = () => {};
+
+      await firebase
```

Comment:
What's this actually testing?

Reply:
This covers all event emitters can be added to the function, helps with code coverage but could probably do with some kind of  assertion that it has worked?

Reply:
Yeah if we actually test they work that'd be good. Right how it's just causing code coverage to increase which isn't really a good idea.

Reply:
Test updated to resolve on a successful emit